specifyweb/frontend/js_src/lib/components/QueryBuilder/Header.tsx

```diff
@@ -79,6 +80,11 @@ export function QueryHeader({
 
   const [isBasic, setIsBasic] = useQueryViewPref(query.id);
 
+  const [showMappingView = true, _] = useCachedState(
```

Comment:
you don't need to pass down `onToggleMapping` at all

the 2nd argument of `useCachedState` in here is the callback to change the state - thus you can change the state from here, without the need for any props from the parent

remember that changes to `useCachedState` are synchronized between all the places where `useCachedState` is used